api/core/workflow/nodes/variable_aggregator/variable_aggregator_node.py

```diff
@@ -21,10 +21,13 @@ def _run(self) -> NodeRunResult:
             for selector in node_data.variables:
                 variable = self.graph_runtime_state.variable_pool.get_any(selector)
                 if variable is not None:
-                    outputs = {"output": variable}
-
-                    inputs = {".".join(selector[1:]): variable}
-                    break
+                    if "output" not in outputs:
+                        outputs["output"] = []
+                    if isinstance(variable, list):
+                        outputs["output"].extend(variable)
+                    elif isinstance(variable, str):
+                        outputs["output"].append(variable)
+                    inputs[".".join(selector[1:])] = variable
```

Comment:
Would you pls explain the purpose of this codes?

Reply:
When the parallel branch calls the knowledge recall and then aggregates, the variable is a list, and the parameter aggregation variable after the ordinary if else node is a str.

Reply:
> When the parallel branch calls the knowledge recall and then aggregates, the variable is a list, and the parameter aggregation variable after the ordinary if else node is a str.

Could you explain the problem you’re encountering? I'd like to understand the context and intent.
If possible, pls create an issue and link it to this PR.